MakerTracker/Controllers/TransactionsController.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using MakerTracker.DBModels;
+using MakerTracker.Models;
+
+namespace MakerTracker.Controllers
+{
+    public class TransactionsController : Controller
+    {
+        private readonly MakerTrackerContext _context;
+
+        public TransactionsController(MakerTrackerContext context)
+        {
+            _context = context;
+        }
+
+        // GET: Transactions
+        public async Task<IActionResult> Index()
+        {
+            var transactions = await _context.Transaction
```

Comment:
Have this return a list of TransactionViewModel. Check out the admin reports index controller for doing a custom select. One of the benefits is that you're not pulling back more info than you need and removes the need for Include()'s and AsNoTracking() (since your selecting into a custom object it knows not to track and also knows to navigate the joins)

Reply:
```suggestion
            var transactions = await _context.Transaction.Select(t=> new TransactionViewModel { Name = t.Firstname, etc etc}).ToListAsync()
```